services/wireguard/wireguard.go

```diff
@@ -73,14 +73,24 @@ type PeerInfo interface {
 	PublicKey() string
 }
 
+// ConsumerPublicKey is used for sending the public key from consumer to provider
+type ConsumerPublicKey struct {
```

Comment:
Bad naming, wg configuration is not about public key only, but can have different fields. Suggesting WireguardConsumerConfig